test/plugins/adonis.spec.js

```diff
@@ -0,0 +1,386 @@
+'use strict'
+
+const axios = require('axios')
+const getPort = require('get-port')
+const agent = require('./agent')
+const plugin = require('../../src/plugins/adonis')
+const { setupResolver, Config, Logger } = require('@adonisjs/sink')
+const { ioc } = require('@adonisjs/fold')
+const path = require('path')
+const frameworkPath = '../../versions/@adonisjs/framework'
+const middlewarePath = '../../versions/@adonisjs/middleware-base'
+const semver = require('semver')
+
+wrapIt()
+
+describe('Plugin', () => {
+  let Server
+  let Route
+  let RouteStore
+  let Request
+  let Response
+  let Context
+  let Exception
+
+  let port
+  let server
+
+  describe('adonis', () => {
+    withVersions(plugin, '@adonisjs/framework', version => {
+      if (semver.intersects(version, '>=5.0.3')) {
```

Comment:
I think this condition can be moved within a single before hook and only determine which `Context.response` getter contents to use and whether to make the `ioc.autoload()` call.  This would remove the redundant imports.

Reply:
Refactoring the code this way leads to the code throwing errors.

Reply:
Figured it out. I still need to have a couple of imports in the `before()`, but I was able to move others out.